scripts/convert_publish.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env python3
+from collections import defaultdict
+import json
+import utils
+import dbs
+import bson
+import sys, traceback
+from datetime import datetime
+from datetime import timedelta
+from decimal import Decimal
+from bson.decimal128 import Decimal128
+from sshtunnel import SSHTunnelForwarder
+from pymongo import MongoClient
+
+def create_tags(metadata_tags):
+    tags = []
+    for tag in metadata_tags or []:
+        tag_obj = {
+            "tag_name": tag
+        }
+        tags.append(tag_obj)
+
+    return tags
+
+class PublishConverter:
+    def __init__(self, golos_db, publish_db, cyberway_db, cache_period = 10000):
+        self.golos_db = golos_db
+        self.publish_db = publish_db
+        self.cyberway_db = cyberway_db
+        self.exists_accs = set()
+        self.mssgs_curatorsw = defaultdict(int)
+        self.publish_tables = dbs.Tables([
+            ('vote',    self.publish_db['votetable'],    None, None),
+            ('message', self.publish_db['messagetable'], None, None),
+            ('content', self.publish_db['contenttable'], None, None)
+        ])
+        self.cache_period = cache_period
+
+    def fill_exists_accs(self):
+        cw_accounts = self.cyberway_db['account']
+        for doc in cw_accounts.find():
+            self.exists_accs.add(doc["name"])
+        print("accs num = ", len(self.exists_accs))
+        for a in self.exists_accs:
+            print(a)
+
+    def convert_posts(self, query = {}):
+        print("convert_posts")
+        golos_gpo = self.golos_db['dynamic_global_property_object'].find()[0]
+        golos_posts = self.golos_db['comment_object']
+        
+        reward_pools = self.publish_db['rewardpools']
+        pool = reward_pools.find()[0] #we have to create it before converting
+        pool["state"]["funds"]["amount"] = utils.get_golos_asset_amount(golos_gpo["total_reward_fund_steem_value"])
+        pool["state"]["funds"]["decs"] = utils.golos_decs
+        pool["state"]["funds"]["sym"] = "GOLOS"
```

Comment:
Symbol and accuracy is also used when converting accounts and balances. I suggest to put them in a separate file with the settings.

Reply:
thx. done